### PR TITLE
Adds AppVeyor and additional information for Jenkins in "Pull Request" at the contributing guide

### PR DESCRIPTION
**Pull request by HyukjinKwon**
This PR proposes to describe AppVeyor with SparkR and some more information related with Jenkins about triggering the test.

I have seen few guys confused about this before:

As a reference for the former - `https://github.com/apache/spark/pull/18164#issuecomment-305734419`
As a reference for the latter - `https://github.com/apache/spark/pull/18444#issuecomment-312468983`

I was also confused when I started to contribute on Spark few years ago.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @srowen and @felixcheung what do you think about this? I am willing to rather close this if you both are not fond of this change. I don't strongly feel about this.

**Comment by srowen** [comment]
Thanks @HyukjinKwon this one is merged too, but I think you'll have to close it manually because I rebased it on top of your other commit.

File: contributing.md [hunk 1]
```diff
@@ -295,8 +295,12 @@ with a link to the full results on Jenkins.
...
      1. Jenkins will automatically re-test when new commits are pushed
      1. If the tests failed for reasons unrelated to the change (e.g. Jenkins outage), then a 
-     committer can request a re-test with "Jenkins, retest this please". 
-     Ask if you need a test restarted.
+     committer can request a re-test with "Jenkins, retest this please".
+     Ask if you need a test restarted. If you were added by "Jenkins, add to whitelist" from a
+     committer before, you can also request the re-test.
+1. If there is a change related to SparkR in your pull request, AppVeyor will be triggered
+automatically to test SparkR on Windows, which takes roughly an hour. Similarly to the steps
+above, fix failures and push new commits which will request the re-test in AppVeyor.
```
Comment (felixcheung) [inline contributing.md:303]: doesn't this need to be indented?

Comment (felixcheung) [reply contributing.md:303]: I'd add to trigger re-test manually, close/re-open PR

**Review by felixcheung — COMMENTED** [review]
sounds good to me to have

File: contributing.md (hunk 1 above)
Comment (HyukjinKwon) [reply contributing.md:303]: Oh, I did not document this as it sounds a rather workaround and I was thinking we should fix this first ... (e.g., opening a INFRA jira and giving committers the right to retrigger ...)

Comment (HyukjinKwon) [reply contributing.md:303]: Will address this one in any event in near future.

**Pull request closed without merging**